### DISCUSSION TITLE
Split provisioning into platform dependant/independant archives

### DISCUSSION THREAD

**Discussion by bublath** [AsamK/signal-cli#498] [Ideas]
Currently the platform depending libraries are for x86.
I'm supporting a plugin to a home automation suite (FHEM) that enables the use of Signal using dbus signal-cli.
See: https://github.com/bublath/FHEM-Signalbot
Since many users struggle, I also created an installation script, that will automatically do all installations, settings etc. to get a working environment. Especially of course for ARM platforms (Raspberry Pi is major platform used for this system).
Currently I provide the armv7 libraries via my private homepage.
I would rather vote for separating the platform depending part (especially now its becoming two libraries) and provide the .so libraries separately.
I'm happy to contribute the armv7 ones, but it should be an aligned process and place to put and download them

**Comment by AsamK** [comment]
Yeah, something automated would be nice, maybe with github actions that upload the assets to a release.
Though I don't have time to to look into it.

For archlinux I already provide AUR packages that build the required native library: https://aur.archlinux.org/packages/signal-cli/

**Comment by bublath** [comment]
For Rapsberry armv7l the libraries required for signal-cli 0.8.0 can be found on my signal-cli based Perl project:
https://github.com/bublath/FHEM-Signalbot/tree/main/armv7l
(No warranty, but they work for me)
